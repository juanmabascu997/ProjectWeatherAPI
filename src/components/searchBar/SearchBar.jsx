import React, { useState } from "react";
import AppBar from "@mui/material/AppBar";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import TextField from "@mui/material/TextField";
import SearchIcon from "@mui/icons-material/Search";
import Box from "@mui/material/Box";
import Container from "@mui/material/Container";
import { IconButton } from "@mui/material";

export default function SearchBar({ onSearch, loading }) {
  const [city, setCity] = useState("");

  return (
      <AppBar
        position="static"
        sx={{ backgroundColor: "#1b1c1d", border: "none", boxShadow: "none" }}
      >
        <Container 
          maxWidth="xl"
          sx={{
            padding: { xs: 0, sm: 0, md: 0, lg: 0, xl: 0 },
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
            justifyContent: 'center',
          }}
        >
          <Toolbar>
            <Typography
              variant="body1"
              component="p"
              sx={{
                flexGrow: 1,
                paddingRight: "5px",
                display: {
                  xs: "none",
                  sm: "block",
                  md: "block",
                },
              }}
            >
              City Search
            </Typography>
            <Box
              component="form"
              noValidate
              autoComplete="off"
              sx={{
                backgroundColor: "dark",
                display: "flex",
                alignItems: "center",
              }}
            >
              <TextField
                id="city-input"
                variant="outlined"
                value={city}
                onChange={(event) => setCity(event.target.value)}
                size="small"
                margin="none"
                sx={{
                  backgroundColor: "dark",
                  borderRadius: 1,
                  mr: 1,
                  input: { color: "white" },
                }}
              />
              <IconButton
                type="submit"
                color="primary"
                id="search-button"
                disabled={loading}
                onClick={(event) => {
                  event.preventDefault();
                  onSearch(city);
                  setCity("");
                }}
              >
                <SearchIcon />
              </IconButton>
            </Box>
          </Toolbar>
        </Container>
      </AppBar>
  );
}
