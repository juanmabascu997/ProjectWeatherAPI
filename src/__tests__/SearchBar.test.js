import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from '../components/searchBar/SearchBar';

describe('SearchBar Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with default props', () => {
    const tree = renderer.create(<SearchBar onSearch={()=>{}} loading={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
