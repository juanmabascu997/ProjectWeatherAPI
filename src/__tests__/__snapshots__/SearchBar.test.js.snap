// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component renders correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1x3x4ku-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-1clzh9g-MuiContainer-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1p8tx8o-MuiTypography-root"
      >
        City Search
      </p>
      <form
        autoComplete="off"
        className="MuiBox-root css-jwayho"
        noValidate={true}
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1undkfv-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="city-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-ihdtdm"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          id="search-button"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
            />
          </svg>
        </button>
      </form>
    </div>
  </div>
</header>
`;

exports[`SearchBar Component renders with default props 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1x3x4ku-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-1clzh9g-MuiContainer-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-1p8tx8o-MuiTypography-root"
      >
        City Search
      </p>
      <form
        autoComplete="off"
        className="MuiBox-root css-jwayho"
        noValidate={true}
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1undkfv-MuiFormControl-root-MuiTextField-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="city-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-ihdtdm"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1kuq5xv-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          id="search-button"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</header>
`;
